src/js/validator.js

```diff
@@ -0,0 +1,50 @@
+const Validate = {
+    validResult: { validated: true, warnings: [] },
+
+    validateForm(form) {
+        this.validResult = { validated: true, warnings: [] };
+        if (form.title.length < 1 || form.title.length > 250)
+            this.updateResult('tutuł nie może mieć mniej niż 1 znaku oraz więcej niż 250');
+        if (form.questions.length < 1)
+            this.updateResult('tutuł nie może mieć mniej niż 1 znaku oraz więcej niż 250');
+        for (const question in form.questions) {
+            if (Object.prototype.hasOwnProperty.call(form.questions, question)) {
+                if (question.type !== 'O' || question.type !== 'W' || question.type !== 'L')
+                    this.updateResult(`niepoprawny typ pytania numer ${question.number}`);
+                if (question.language !== 'EN' || question.language !== 'PL')
+                    this.updateResult(`niepoprawny język pytania numer ${question.number}`);
+                if (question.content.length < 1 || question.content.length > 250)
+                    this.updateResult(`zawartość pytania numer ${question.number} nie może być mniejsza niż 1 oraz więcej niż 250`);
+                if (question.type === 'W' || question.answers.length < 1)
+                    this.updateResult('pytania zamknięte nie mogą mieć mniej niż 2 odpowiedzi');
+                if (question.type === 'W') {
+                    for (const answer in question.answers) {
+                        if (answer.length < 1 || answer.length > 250)
+                            this.updateResult(`odpowiedz pytania zamknietego o numerze ${question.number} nie może mieć mniej niż 1 znak oraz więcej niż 250`);
+                    }
+                }
+            }
+        }
+        return this.validResult;
+    },
+    validateFilledForm(filledForm) {
+        this.validResult = { validated: true, warnings: [] };
+        if (filledForm.owner.length < 1)
+            this.updateResult('błąd podczas odczytywania nazwy użytkownika');
+        for (const question in filledForm.questions) {
+            if (Object.prototype.hasOwnProperty.call(filledForm.questions, question)) {
+                if (question.userAnswer.length < 1 || question.userAnswer.length > 250)
+                    this.updateResult(`odpowiedz pytania zamknietego o numerze ${question.number} nie może mieć mniej niż 1 znak oraz więcej niż 250`);
```

Comment:
Rozbił bym to na dwa: 1. nie może być pusty;  2. nie może mieć więcej niż 250 znaków, aczkolwiek tak też jest spoko